### PR TITLE
Allows BR & MX admins to edit any content

### LINKED ISSUE #5020: BR/MX admin need to to be able to edit content not created by them
We'll need to add these permissions to BR/MX admin roles:
- [ ] Campaign: Edit any content
- [ ] Static content: Edit any content

keep in mind though, that Brazil admins should only have access to Brazilian Portuguese fields and Mexico admins should only have access to Mexican Spanish fields from @blisteringherb's [comment](https://github.com/DoSomething/phoenix/issues/4934#issuecomment-136467741) on #4934:

"I'm thinking that putting in a hook_menu_alter for node/%/edit that triggers the add translation chain if the allowed translation hasn't yet been created for that node and if it already exists it will send the user to node/%/edit/es-mx so they never actually get to the edit screen for the non-allowed translations.

For instance if a user with the Brazil admin role tries to visit `node/36/edit`, they'll be redirected to `node/36/edit/pt-br` if the translation already exists, and if not they'll go to `node/57/edit/add/en/pt-br`.  The other languages in the upper right corner of the node edit screen will be hidden except for those that they're allowed to see, and they won't have access to the translate tab."

note: whether or not they need access to `Home: Edit any content` is dependent on the decision made in #4968

### PR DESCRIPTION
**Pull request by itsjoekent**
#### What's this PR do?
- Exports new regional admin permissions to edit any campaign content and static content
- Redirects users to the proper translation page if the node source language does not match the user language 
#### Where should the reviewer start?

dosomething_campaign.module -- campaign_init()
#### How should this be manually tested?
- Create a new campaign 
- Under a new account with a different language, edit the campaign. 
  - It should redirect you to a url containing /add/<node language>/<user language>
- Save the campaign
- Hit 'Edit' again
  - It should redirect you to a url ending with your user language 
#### Any background context you want to provide?

Multiple hooks were tried and discussed in #global but in the end we decided to go with hook_init() 
#### Relevant tickets

Fixes #5020 

@angaither

### CODE REVIEW

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.strongarm.inc [hunk 1]
```diff
@@ -16,7 +16,7 @@ function dosomething_campaign_strongarm() {
   $strongarm->name = 'entity_translation_settings_node__campaign';
   $strongarm->value = array(
     'default_language' => 'xx-et-author',
-    'hide_language_selector' => 1,
+    'hide_language_selector' => 0,
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.strongarm.inc]: I think we still want to keep this hidden

**Comment by angaither** [comment]
@deadlybutter when I pulled this down it worked for brazilian admins, but when I went to edit it as a regular admin it was adding the translations stuff and redirecting me here http://dev.dosomething.org:8888/node/1485/edit/add/en/
which 404s

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.strongarm.inc (hunk 1 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.strongarm.inc]: Im not 100% what this is and I don't remember changing it?

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module [hunk 2]
```diff
@@ -26,6 +26,37 @@ function dosomething_campaign_form_campaign_node_form_alter(&$form, &$form_state
   $form['#after_build'][] = 'dosomething_campaign_form_campaign_node_form_after_build';
 }
 
+function dosomething_campaign_init() {
+  // Get the URL and split it up
+  $current_path = current_path();
+  $path_items = explode('/', $current_path);
+
+  // Verify we're dealing with a node edit with no translation specification in the URL
+  if ($path_items[0] == "node" && is_numeric($path_items[1]) && count($path_items) == 3) {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module]: this catches all those node tabs at the top
`node/[id]/revisions`
`node/[id]/pitch`
`node/[id]/custom-settings`
`node/[id]/rb`
and redirects to the edit page

**Comment by itsjoekent** [comment]
Yeah that URL looks wrong,

what language was the campaign created in?

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module (hunk 2 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module]: oh forgot about those, Ill also verify the 3rd param is /edit

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.strongarm.inc (hunk 1 above)
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.strongarm.inc]: it's the language selector at the top of the nodes
screen shot 2015-09-10 at 1 00 34 pm

**Comment by angaither** [comment]
@deadlybutter english

**Comment by itsjoekent** [comment]
thanks, looking into it now!

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module [hunk 3]
```diff
@@ -26,6 +26,37 @@ function dosomething_campaign_form_campaign_node_form_alter(&$form, &$form_state
   $form['#after_build'][] = 'dosomething_campaign_form_campaign_node_form_after_build';
 }
 
+function dosomething_campaign_init() {
+  // Get the URL and split it up
+  $current_path = current_path();
+  $path_items = explode('/', $current_path);
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module]: `arg()` https://api.drupal.org/api/drupal/includes%21bootstrap.inc/function/arg/7

Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module]: arrrgggg

image

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module [hunk 4]
```diff
@@ -26,6 +26,37 @@ function dosomething_campaign_form_campaign_node_form_alter(&$form, &$form_state
   $form['#after_build'][] = 'dosomething_campaign_form_campaign_node_form_after_build';
 }
 
+function dosomething_campaign_init() {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module]: two things here
1. it's typically standard to have `init` hooks at the top of the module file
2. maybe this should live in `dosomething_global`

**Comment by itsjoekent** [comment]
Note: this commit https://github.com/deadlybutter/dosomething/commit/478708ca7ab3ef326bb3fbae9e6c1f3baf631d98

fixed the bug Andrea raised above ^^

We logged the user_language and it wasn't on her account, thus the URL was a 404 because it was missing data. For the moment I made the default here 'en' but when the strongarm variables are finished this can become more dynamic / based on user role.

**Comment by angaither** [comment]
@deadlybutter actually @blisteringherb merged the strongarm stuffs in #5068

**Comment by itsjoekent** [comment]
oh, great!

**Comment by angaither** [comment]
@deadlybutter might be worth making a new PR (perhaps before this one) to update `dosomething_global_get_user_language()`

**Comment by itsjoekent** [comment]
Hmm, I dont have that function, Im still out of date

**Comment by itsjoekent** [comment]
@angaither took care of all the comments I think -- should I work on merging this first? (otherwise it might start to get confusing) after that I'll rebase my code and update all the user language logic in a new PR

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module [hunk 5]
```diff
@@ -14,6 +14,33 @@ include_once 'includes/CampaignTransformer.php';
 include_once 'dosomething_campaign.theme.inc';
 define('DOSOMETHING_CAMPAIGN_PIC_STEP_HEADER', 'Snap a Pic');
 
+function dosomething_campaign_init() {
```
Comment (angaither) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module]: I think we should probably still have this live in `dosomething_global` module

**Comment by angaither** [comment]
bahh, got a redirect loop on `node/edit` as a regional admin

**Comment by itsjoekent** [comment]
fuckin redirect loops

File: lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module (hunk 5 above)
Comment (itsjoekent) [inline lib/modules/dosomething/dosomething_campaign/dosomething_campaign.module]: whooops missed this one. yeah, thats a good point as its more global logic than campaign logic.

**Comment by itsjoekent** [comment]
@angaither I think we're good now

**Comment by itsjoekent** [comment]
wait we're not. something happened to the git history. this looks like the same thing that just happened to @weerd 

I havent rebased or done anything I can think of that could have caused that to happen

**Pull request closed without merging**